### PR TITLE
Feature/creating data model

### PR DESCRIPTION
**Pull request by waelelsafty07**
This PR is raised to implement the following tasks:

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
   - Create and run the necessary migration files.
          - Table and column names should match the ERD diagram.

   - Foreign keys should be included.

### CODE REVIEW

File: db/migrate/20231027125350_create_likes.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :author_id, foreign_key: { to_table: :users }
+      t.references :post_id, foreign_key: { to_table: :posts }
```
Comment (elisha2kyakpo1) [inline db/migrate/20231027125350_create_likes.rb:5]: - [ ] Let's make sure the foreign keys (user) in the comments and likes tables (migration files) match the ERD. The only table to have `author_id` as foreign key is a posts table. This table should reference a user.

File: db/migrate/20231027125416_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :author_id, foreign_key: { to_table: :users }
+      t.references :post_id, foreign_key: { to_table: :posts }
```
Comment (elisha2kyakpo1) [inline db/migrate/20231027125416_create_comments.rb:5]: - [ ] Let's make sure the foreign keys (user) in the comments and likes tables (migration files) match the ERD. The only table to have `author_id` as foreign key is a posts table. This table should reference a user.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @waelelsafty07, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
👏 The posts migration data model is set up nicely.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Strangeal — APPROVED** [review]
Hi @waelelsafty07, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Added `user` reference ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**